src/Features/Core/Portable/MoveToNamespace/AbstractMoveToNamespaceService.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.MoveToNamespace
+{
+    internal interface IMoveToNamespaceService : ILanguageService
+    {
+        Task<ImmutableArray<AbstractMoveToNamespaceCodeAction>> GetCodeActionsAsync(Document document, TextSpan span, CancellationToken cancellationToken);
+        Task<MoveToNamespaceAnalysisResult> AnalyzeTypeAtPositionAsync(Document document, int position, CancellationToken cancellationToken);
+        Task<MoveToNamespaceResult> MoveToNamespaceAsync(MoveToNamespaceAnalysisResult analysisResult, string targetNamespace, CancellationToken cancellationToken);
+        MoveToNamespaceOptionsResult GetChangeNamespaceOptions(Document document, string defaultNamespace, ImmutableArray<string> namespaces);
+        bool IsValidIdentifier(string identifier);
```

Comment:
This feels out of place. Perhaps pass in SyntaxFacts to this type and use it directly instead?

Reply:
SyntaxFacts is a concrete class depending on what language is being used, which is why I abstracted to here. If it was an interface I would just use that

Reply:
Can you use ISyntaxFactsService: http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/LanguageServices/SyntaxFactsService/ISyntaxFactsService.cs,6808a5092d840942,references

Reply:
I swear I was looking for that and couldn't find it. Seems perfect